packages/web-app-files/tests/unit/components/AppBar/ViewOptions.spec.ts

```diff
@@ -33,9 +34,28 @@ describe('ViewOptions component', () => {
     wrapper.find(selectors.fileExtensionsSwitch).vm.$emit('change', false)
     expect(storeOptions.modules.Files.mutations.SET_FILE_EXTENSIONS_VISIBILITY).toHaveBeenCalled()
   })
+  it('initially does not show a viewmode switcher', () => {
+    const { wrapper } = getWrapper()
+    expect(wrapper.find('[data-testid="viewmode-switch-buttons"]').exists()).toBeFalsy()
+  })
+  it('shows a viewmode switcher if more than one viewModes are passed', () => {
+    const { wrapper } = getWrapper(
+      {},
+      {
+        viewModes: [ViewModeConstants.condensedTable, ViewModeConstants.default]
+      }
+    )
+    const viewModeSwitchButtons = wrapper.find('[data-testid="viewmode-switch-buttons"]')
+    expect(viewModeSwitchButtons).toMatchSnapshot()
+  })
 })
 
-function getWrapper({ perPage = '100' } = {}) {
+function getWrapper(
+  { perPage = '100' } = {},
+  props: { [key: string]: any } = {
```

Comment:
No need for this type

Reply:
VSCode wanted one pretty badly, but I'll try without...